src/library/scala/collection/immutable/ChampHashMap.scala

```diff
@@ -553,6 +557,26 @@ private final class BitmapIndexedMapNode[K, +V](
     }
   }
 
+  override def transform[W](f: (K, V) => W): BitmapIndexedMapNode[K, W] = {
+    val newContent = content.clone()
+    val _payloadArity = payloadArity
+    val _nodeArity = nodeArity
+    val newContentLength = newContent.length
+    var i = 1
+    while (i < _payloadArity) {
+      newContent(TupleLength * i + 1) = f(getKey(i), getValue(i))
```

Comment:
Could also detect if `f(...) eq getValue(i)` for all elements, and similarly if all the `getNode(i).transfom()`-s are identities, and then just return `this` rather than the new node.

Reply:
So we traverse the whole trie, and build up a new one, but then at the very end if identical, return `this`?

Reply:
That's what I had in mind, it doesn't avoid the allocation but does reduce the footprint afterwards.

I guess we could lazily create `newContent` on the first non-identity transform.

Reply:
Something like:

```scala
  override def transform[W](f: (K, V) => W): BitmapIndexedMapNode[K, W] = {
    var newContent: Array[Any] = null
    val _payloadArity = payloadArity
    val _nodeArity = nodeArity
    val newContentLength = content.length
    var i = 1
    while (i < _payloadArity) {
      val key = getKey(i)
      val value = getValue(i)
      val newValue = f(key, value)
      if (newContent eq null) {
        if (newValue.asInstanceOf[AnyRef] ne value.asInstanceOf[AnyRef]) {
          newContent = content.clone
          newContent(TupleLength * i + 1) = newValue
        }
      } else {
        newContent(TupleLength * i + 1) = newValue
      }
      i += 1
    }

    var j = 0
    while (j < _nodeArity) {
      val node = getNode(j)
      val newNode = node.transform(f)
      if (newContent eq null) {
        if (newNode ne node) {
          newContent = content.clone()
          newContent(newContentLength - j - 1) = newNode
        }
      } else
        newContent(newContentLength - j - 1) = newNode
      j += 1
    }
    if (newContent eq null) this.asInstanceOf[BitmapIndexedMapNode[K, W]]
    else new BitmapIndexedMapNode[K, W](dataMap, nodeMap, newContent, originalHashes, size)
  }
```

Reply:
![woah](https://media.giphy.com/media/xT0xeJpnrWC4XWblEk/giphy.gif)